packages/stark-ui/src/modules/table/components/table.component.ts

```diff
@@ -341,8 +341,13 @@ export class StarkTableComponent extends AbstractStarkUiComponent implements OnI
 	 */
 	// tslint:disable-next-line:cognitive-complexity
 	public ngOnChanges(changes: SimpleChanges): void {
-		if (changes["data"] && !changes["data"].isFirstChange() && this.resetFilterValueOnDataChange()) {
-			this.applyFilter();
+		if (changes["data"] && !changes["data"].isFirstChange()) {
+			this.resetFilterValueOnDataChange();
+
+			// When data changes, we should sort it again in any case because:
+			//   * orderProperties could contain some sort rules
+			//   * every column could have a sort rule defined
+			this.sortData();
```

Comment:
True, we should re-sort the data, but we should also call `applyFilter()` as before... the `resetFilterValueOnDataChange()` method returns a boolean that indicates whether the filter has been reset and in that case we should re-filter the data ;)